### PR TITLE
Make positive shape point/corner X values count as 'forward'

### LINKED ISSUE #14990: HitShape point/corner X value having reverse effect is counter-intuitive
In pretty much all other places where we define offsets (including `TargetableOffsets` on `HitShape` trait), a positive X value means "forward" (in the direction the actor is facing, in case of immobile actors like buildings: north).

However, on non-Circle hit shape types, a positive X value means backwards/south, while a negative value means forward/north.
This is confusing for modders, even I have occasionally had trouble dealing with that, despite being 'in the know'.

### PR DESCRIPTION
**Pull request by reaperrr**
Closes #14990.

To keep the regression risk as low as possible, I went for only changing the modder-facing value and internally convert it back to what the math formulas expect.

Note: Update rule has been tested and used to update all shipping mods.
Update rule for Polygon shape has not been included because a) I couldn't get it to work on an int2 array and b) the Polygon shape was merged after last release, so modders updating from release to release won't need an update rule for that shape anyway.

### CODE REVIEW

**Comment by reaperrr** [comment]
@abcdefg30 @pchote Unless there's a technical reason against my approach, I'd prefer this to make it into Next as well, so I can avoid having to write an update rule for the Polygon shape.

**Review by GraionDilach — APPROVED** [review]
:+1:

**Comment by pchote** [comment]
There is definitely something technically weird going on that I would like to understand rather than (or at least before) bandaiding over.  If these values are being used sensibly then I don't understand how their sign could become inverted, unless something is accidentally being subtracted instead of added.  If that is the case then it would be better to fix the broken calculation than to change it at the input.

File: OpenRA.Mods.Common/UpdateRules/Rules/SwitchShapePointsX.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class SwitchShapePointsX : UpdateRule
+	{
+		public override string Name { get { return "Switch positive and negative X values on HitShape points"; } }
```
Comment (pchote) [inline OpenRA.Mods.Common/UpdateRules/Rules/SwitchShapePointsX.cs]: Are you 100% sure that `Y` isn't inverted too?

File: mods/cnc/rules/civilian-desert.yaml [hunk 2]
```diff
@@ -4,8 +4,8 @@ V20:
 		UseTargetableCellsOffsets: false
 		TargetableOffsets: -840,-512,0, 0,0,0, -840,512,0
 		Type: Rectangle
-			TopLeft: -1024, -512
-			BottomRight: 1024, 896
+			TopLeft: 1024,-512
+			BottomRight: -1024,896
```
Comment (pchote) [inline mods/cnc/rules/civilian-desert.yaml]: This puts the point to the left of the `TopLeft`.  The rule really should switch `TopLeft` and `BottomRight` (which could be a straight rename if `Y` indeed also does need to be inverted).

File: OpenRA.Mods.Common/UpdateRules/Rules/SwitchShapePointsX.cs (hunk 1 above)
Comment (reaperrr) [reply OpenRA.Mods.Common/UpdateRules/Rules/SwitchShapePointsX.cs]: Not in terms of end result in any case, on buildings (which are always facing 0/north), negative means "left", which matches how all our other offsets work too, afaik.

**Pull request closed without merging**